### PR TITLE
Remove last of Cry navigation

### PR DESCRIPTION
**Pull request by greerdv**
## What does this PR do?
Removes the last bits of cry navigation

## How was this PR tested?
Built and ran tests.

### CODE REVIEW

**Review by AMZN-daimini — APPROVED** [review]

File: Code/Legacy/CryCommon/crycommon_files.cmake [hunk 1]
```diff
@@ -28,7 +28,7 @@ set(FILES
     IValidator.h
     IWindowMessageHandler.h
     IXml.h
-    INavigationSystem.h
+    MicrophoneBus.h
```
Comment (lumberyard-employee-dm) [inline Code/Legacy/CryCommon/crycommon_files.cmake:31]: I am guessing the MicrophoeBus.h header was missing from the list of CryCommon headers?

Comment (greerdv) [reply Code/Legacy/CryCommon/crycommon_files.cmake]: There was a merge conflict when I merged from development and I messed up resolving it. I've just fixed it.

**Review by lumberyard-employee-dm — APPROVED** [review]

**Review by moraaar — APPROVED** [review]
Thanks for removing this.

**Review by cgalvan — APPROVED** [review]
Nice! Can you also remove the component icons for these?

`Assets\Editor\Icons\Components\Navigation*.svg`

**Comment by greerdv** [comment]
> Nice! Can you also remove the component icons for these?
> 
> `Assets\Editor\Icons\Components\Navigation*.svg`

Forgot about those, thanks for spotting. I've already started an AR run so I'll try and get rid of those in another change if that's OK.

**Comment by hultonha** [comment]
Awesome work mopping this up @greerdv, all these small incremental removals/clean-ups make a big difference over time!

**Pull request merged**